Sources/MongoSwift/ReadWriteConcern.swift

```diff
@@ -0,0 +1,104 @@
+import libmongoc
+
+/// An enumeration of possible ReadConcern levels.
+public enum ReadConcernLevel: String {
+    case local
+    case available
+    case majority
+    case linearizable
+    case snapshot
+}
+
+/// A class to represent a MongoDB read concern.
+public class ReadConcern: Equatable, CustomStringConvertible {
+
+    /// A pointer to a mongoc_read_concern_t
+    internal var _readConcern: OpaquePointer?
+
+    /// The level of this readConcern, or nil if the level is not set.
+    public var level: String? {
+        guard let level = mongoc_read_concern_get_level(self._readConcern) else {
+            return nil
+        }
+        return String(cString: level)
+    }
+
+    private var asDocument: Document {
+        let doc = Document()
+        try? self.append(to: doc)
+        return doc
+    }
+
+    public var description: String {
+        return self.asDocument.description
+    }
+
+    /// Initialize a new ReadConcern from a ReadConcernLevel.
+    public convenience init(_ level: ReadConcernLevel) throws {
+        try self.init(level.rawValue)
+    }
+
+    /// Initialize a new ReadConcern from a String corresponding to a read concern level.
+    public init(_ level: String) throws {
+        self._readConcern = mongoc_read_concern_new()
+        if !mongoc_read_concern_set_level(self._readConcern, level) {
+            throw MongoError.readConcernError(message: "Failed to set read concern level to '\(level)'")
+        }
+    }
+
+    /// Initialize a new empty ReadConcern.
+    public init() {
```

Comment:
does it make sense to make a copy initializer for read concerns? like `init(fromReadConcern: ReadConcern)`

Reply:
sure, added